plugins/fossa/src/api/FossaClient.ts

```diff
@@ -37,42 +52,101 @@ export class FossaClient implements FossaApi {
     this.organizationId = organizationId;
   }
 
-  private async callApi(path: string): Promise<any> {
+  private async callApi<T>(
+    path: string,
+    query: Record<string, any>,
+  ): Promise<T | undefined> {
     const apiUrl = `${await this.discoveryApi.getBaseUrl('proxy')}/fossa`;
     const idToken = await this.identityApi.getIdToken();
-    const response = await fetch(`${apiUrl}/${path}`, {
-      headers: idToken ? { Authorization: `Bearer ${idToken}` } : {},
-    });
-    if (response.status === 200) {
-      return await response.json();
+    const response = await fetch(
+      `${apiUrl}/${path}?${new URLSearchParams(query).toString()}`,
+      {
+        headers: idToken ? { Authorization: `Bearer ${idToken}` } : {},
+      },
+    );
+
+    if (!response.ok) {
+      throw await ResponseError.fromResponse(response);
     }
-    return undefined;
+
+    return (await response.json()) as T;
   }
 
-  async getFindingSummary(
-    projectTitle: string,
-  ): Promise<FindingSummary | undefined> {
-    const project = await this.callApi(
-      `projects?count=1&sort=title+&title=${projectTitle}${
-        this.organizationId ? `&organizationId=${this.organizationId}` : ''
-      }`,
-    );
+  async *getProject(
+    projectTitles: Set<string>,
+  ): AsyncIterable<{ title: string; summary: FindingSummary }> {
+    let nextPage = 0;
+    const pageSize = 1000;
+
+    for (;;) {
+      const projects = await this.limit(
+        page =>
+          this.callApi<FossaProjectsResponse[]>('projects', {
+            count: pageSize,
+            page,
+            sort: 'title+',
+            ...(this.organizationId && {
+              organizationId: this.organizationId,
+            }),
+            ...(projectTitles.size === 1 && {
+              title: projectTitles.values().next().value,
+            }),
+          }),
+        nextPage++,
+      );
+
+      if (!projects) {
```

Comment:
It's a bit funky that the api call returns undefined for non-200 - feels like it will hide real errors.

Could also do the `projects.length < pageSize` check all the way up here to only have one break condition

Reply:
Good catch! Yes that's a leftover. I refactored `callApi` to throw an exception but haven't changed the return type accordingly.